### PR TITLE
Feature/backend/go vendoring

### PR DESCRIPTION
**Pull request by sapiens-sapide**
Change the place of Go's vendor directory to enable a global Go dependencies management, at repos' level.

### CODE REVIEW

File: src/backend/configs/caliopen-go-lmtp_dev.yaml [hunk 1]
```diff
@@ -24,18 +24,18 @@ AppConfig:
 ## LDA (Email broker) config ##
 LDAConfig:
   broker_type: smtp # types are : smtp, imap, mailboxe, etc.
-  nats_url: nats://localhost:4222
+  nats_url: nats://nats.dev.caliopen.org:4222
```
Comment (iamdey) [inline src/backend/configs/caliopen-go-lmtp_dev.yaml]: this config shouldn't be restricted to a docker environment ?

I mean this is used for both native & vagrant env, right ?

File: src/backend/Dockerfile.go-api [hunk 2]
```diff
@@ -0,0 +1,15 @@
+FROM golang
+
+RUN go get -u github.com/kardianos/govendor
+RUN go install github.com/kardianos/govendor
```
Comment (sapiens-sapide) [inline src/backend/Dockerfile.go-api]: go install should not be necessary, go get already do it.

File: src/backend/Dockerfile.go-api [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+ADD . /go/src/github.com/CaliOpen/CaliOpen/src/backend
+WORKDIR /go/src/github.com/CaliOpen/CaliOpen/src/backend
+
+RUN govendor sync -v
```
Comment (sapiens-sapide) [inline src/backend/Dockerfile.go-api]: to be sure to use the govendor bin inside the docker I'll suggest `$GOPATH/bin/govendor`

File: src/backend/Dockerfile.go-lmtp [hunk 4]
```diff
@@ -0,0 +1,15 @@
+FROM golang
+
+RUN go get -u github.com/kardianos/govendor
+RUN go install github.com/kardianos/govendor
+
+ADD . /go/src/github.com/CaliOpen/CaliOpen/src/backend
+WORKDIR /go/src/github.com/CaliOpen/CaliOpen/src/backend
+
+RUN govendor sync -v
```
Comment (sapiens-sapide) [inline src/backend/Dockerfile.go-lmtp]: `govendor sync` => `$GOPATH/bin/govendor sync`

File: src/backend/configs/caliopen-go-lmtp_dev.yaml (hunk 1 above)
Comment (sapiens-sapide) [inline src/backend/configs/caliopen-go-lmtp_dev.yaml]: yes

**Pull request merged**